src/client/actions/ThreadListSync.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const Action = require('./Action');
+const Collection = require('../../util/Collection');
+const { Events } = require('../../util/Constants');
+
+class ThreadListSyncAction extends Action {
+  handle(data) {
+    const client = this.client;
+
+    const guild = client.guilds.cache.get(data.guild_id);
+    if (!guild) return {};
+
+    if (!data.channels_ids) {
+      guild.channels.cache.forEach(channel => {
+        this.removeStale(channel);
+      });
+    } else {
+      data.channel_ids.forEach(id => {
+        const channel = client.channels.resolve(id);
+        if (channel) this.removeStale(channel);
+      });
+    }
```

Comment:
Worlds smallest nit but can you inverse the if checks? so `if data.channel_ids: removeStaleFromThat(); else: removeAllStaleFromGuild()`